stellargraph/core/element_data.py

```diff
@@ -303,6 +305,37 @@ def _numpyise(d, dtype):
     return {k: np.array(v, dtype=dtype) for k, v in d.items()}
 
 
+class FlatAdjacencyList:
+    def __init__(self, flat_array, neighbor_counts):
+        self.splits = np.cumsum(np.append(0, neighbor_counts))
```

Comment:
I'd suggested the version with `neighbour_counts = np.zeros(number_of_nodes + 1, dtype=dtype)` and slicing to avoid having to do yet another `append` and create yet another (slightly) larger array here.